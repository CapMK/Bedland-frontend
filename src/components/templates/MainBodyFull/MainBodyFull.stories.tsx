import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import { Provider } from 'react-redux';
import { reduxStore } from 'store/store';
import Header from 'components/organisms/Header/Header';
import Footer from 'components/organisms/Footer/Footer';
import MainBodyFull from './MainBodyFull';

const meta = {
  title: 'Templates/MainBodyFull',
  component: MainBodyFull,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <BrowserRouter>
        <ThemeProvider>
          {' '}
          <Provider store={reduxStore}>
            <GlobalStyle />
            <Header />
            {Story()}
            <Footer />
          </Provider>
        </ThemeProvider>
      </BrowserRouter>
    ),
  ],
  parameters: {
    backgrounds: { default: 'dark' },
  },
} satisfies Meta<typeof MainBodyFull>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: (
      <>
        <div
          style={{
            border: '1px solid white',
            height: '300px',
            width: '300px',
            color: 'white',
            backgroundColor: 'darkgrey',
          }}
        >
          left content section
        </div>
        <div
          style={{
            border: '1px solid white',
            height: '300px',
            width: '300px',
            color: 'white',
            backgroundColor: 'darkgrey',
          }}
        >
          right content section
        </div>
      </>
    ),
  },
  argTypes: {
    children: {
      control: false,
    },
  },
};
