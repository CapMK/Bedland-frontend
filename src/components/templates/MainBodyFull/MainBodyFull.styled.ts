import { HexColor } from 'theme/theme.types';
import styled from 'styled-components';

export const MainBodyFullLayout = styled.div<{ backgroundColor: HexColor }>`
  align-items: center;
  display: flex;
  gap: 11.25rem;
  justify-content: center;
  background: ${({ backgroundColor }) => backgroundColor}};
  width: 100%;
`;
export default MainBodyFullLayout;
