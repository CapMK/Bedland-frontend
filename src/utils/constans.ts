export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
