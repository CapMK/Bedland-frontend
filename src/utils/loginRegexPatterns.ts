import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
