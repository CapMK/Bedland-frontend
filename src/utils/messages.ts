const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
