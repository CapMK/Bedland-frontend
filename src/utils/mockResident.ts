const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
