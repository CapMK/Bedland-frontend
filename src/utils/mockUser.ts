import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
