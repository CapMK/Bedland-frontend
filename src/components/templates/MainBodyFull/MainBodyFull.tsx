import { PropsWithChildren, useContext } from 'react';
import { ColorNames } from 'theme/theme.types';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './MainBodyFull.styled';

type MainBodyFullProps = {
  backgroundColor?: ColorNames;
};

const defaultProps: MainBodyFullProps = {
  backgroundColor: 'sidebar',
};

const MainBodyFull = ({
  backgroundColor = 'sidebar',
  children,
}: PropsWithChildren<MainBodyFullProps>) => {
  const { palette } = useContext(ThemeContext);
  return (
    <Styled.MainBodyFullLayout backgroundColor={palette[backgroundColor]}>
      {children}
    </Styled.MainBodyFullLayout>
  );
};

export default MainBodyFull;

MainBodyFull.defaultProps = defaultProps;
