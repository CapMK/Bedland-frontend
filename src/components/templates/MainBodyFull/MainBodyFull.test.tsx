import { render } from 'utils/test.utils';
import { cleanup } from '@testing-library/react';
import MainBodyFull from './MainBodyFull';

describe('Login Page Template', () => {
  const item = () => render(<MainBodyFull />, {});
  afterEach(cleanup);
  it('renders correctly', () => {
    expect(item).toMatchSnapshot();
  });
});
