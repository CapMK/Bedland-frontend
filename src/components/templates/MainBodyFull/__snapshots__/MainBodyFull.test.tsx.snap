// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page Template > renders correctly 1`] = `[Function]`;
